retriever/lib/get_datapackages.py

```diff
@@ -0,0 +1,96 @@
+import os.path
+import yaml
+import json
+from urllib.request import urlopen
+from urllib.parse import urljoin, urlparse, urlunparse, ParseResult
+
+from retriever import HOME_DIR, SCRIPT_SEARCH_PATHS
+
+def get_dps():
+    """Load the list of data packages from datapackages.yml
+
+    Checks all of the search paths for a datapackages.yml file and loads it into
+    a dictionary.
+
+    """
+    for path in SCRIPT_SEARCH_PATHS:
+        try:
+            with open(os.path.join(path, "datapackages.yml"), 'r') as dp_file:
+                dp_dict = yaml.safe_load(dp_file)
+        except:
+            pass
+    return dp_dict
+
+def load_dp_json(url):
+    """Load a remote data package json file"""
+    with urlopen(url) as url:
+        dp = json.loads(url.read().decode())
+    return dp
+
+def replace_path(dp, url):
+    """Replace relative data locations with absolute urls
+
+    Data packages can either use relative paths or absolute urls to identify the
+    locations of data: http://frictionlessdata.io/guides/data-package/#resources
+    Convert all of the relative paths into absolute urls for consistent
+    processing
+
+    """
+    for resource in dp['resources']:
+        if 'path' in resource:
+            resource['url'] = urljoin(url, resource['path'])
+            del resource['path']
+    return dp
+
+def add_retriever_metadata(dp):
+    """Add retriever specific metadata to a data package json file"""
+    dp["retriever"] = "True"
+    dp["retriever_minimum_version"] = "2.0.0"
+    return dp
+
+def replace_name(dp, dp_name):
+    """Replace the original dp name with the name chosen for the retriever
+
+    There is no system for ensuring that data package names are unique and
+    descriptive, so we replace the name in the original dp with the name
+    specified in datapackage.yml
+    """
+    dp["name"] = dp_name
+    return dp
+
+def match_types(dp):
+    """Match the datapackage types to the retriever types
+
+    The retriever still describes types using its old system instead of the
+    official frictionless data spec. This converts frictionless data types to
+    retriever types.
+
+    """
+
+    types = {'string': 'char',
+             'number': 'double',
+             'integer': 'int',
+             'date': 'char'
+    }
```

Comment:
- [ ]  Closing bracket not indented well
- [ ] Double spacing between functions
- [ ] Do we need the extra libs `urlparse, urlunparse, ParseResult`